src/hotspot/share/classfile/modules.cpp

```diff
@@ -628,17 +628,20 @@ void Modules::serialize(SerializeClosure* soc) {
 }
 
 void Modules::dump_native_access_flag() {
+  ResourceMark rm;
   const char* native_access_names = get_native_access_flags_as_sorted_string();
   if (native_access_names != nullptr) {
     _archived_native_access_flags = ArchiveBuilder::current()->ro_strdup(native_access_names);
   }
 }
 
 const char* Modules::get_native_access_flags_as_sorted_string() {
+  assert(Thread::current()->current_resource_mark() != nullptr, "Setup by caller");
   return get_numbered_property_as_sorted_string("jdk.module.enable.native.access");
 }
 
 void Modules::serialize_native_access_flags(SerializeClosure* soc) {
+  ResourceMark rm;
```

Comment:
I would suggest moving this to the scope where it is needed.

Reply:
This is the scope where `ResourceMark` is needed. no?

Reply:
I think what David meant is to declare `ResourceMark` inside the "if" block, right before calling `check_archived_flag_consistency()`. Similarly in `serialize_addmods_names()`.